### ISSUE TITLE
Promise not triggering on Android through wechat

### ISSUE THREAD

**Issue by astik** [dracupid/light-promise#3]
When using the library on wechat on Android (4.4.2), the timeout is not firing. When debugging, we can see that setTimeout is undefined.

By looking at the dist livbrary :

``` js
var STATE, isFun, isObj, redEnd, redStart, setTimeout, unCaughtError,
  bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; };
```

setTimeout is overriden in this configuration and it is set to undefined.
As I know that i am not using nextTick on this website (frontend only), I remove the setTimeout from the declared variable list. Everything went back in order.

Supposed faulty hash : 9a4140c32d61e885af666cdfdcb80d80e9cf1c3e
Version affected : 0.0.4
Platform : Android 4.4.2 + wechat

A solution for the dist library will be to wrap all the code into an IIFE :

``` js
(function(setTimeout){
  ...
}( (process && process.nextTick) || setTimeout));
```

Also this will avoid all variable to go global.

I have to go back from 0.0.4 to 0.0.3 only for this issue ='(

**Comment by astik** [comment]
I am no Coffee expert but finding a way to tell coffee not to add setTimeout to the variable list or that setTimeout is a global would be enough.

**Comment by dracupid** [comment]
It's bad to overwrite a global variable, I have fixed this but not published. Try 0.1.0 (later) and see if it still happens.

**Comment by dracupid** [comment]
I have rewritten this project in ES6, no more coffeescript.

**Comment by astik** [comment]
Glad to hear all of this. Waiting for 0.1.0 then =)

**Comment by astik** [comment]
0.1.0 seems to be available on NPM but there is no associated tag. Anyway, i tried this new version.
Problem with timeout seems to be fixed but, another bug appears.
When trying to access the process variable, the JS engine fail silently. So this line of code cannot be executed :

``` js
var $nextTick = process ? process.nextTick : setTimeout;
```

I tried to display the process value with an alert (i know, this is not a way to do debug but with wechat you may encounter some difficulties ...) :

``` js
alert(process);
```

Just accessing it would lead to the same behavior : the JS engine starts to evaluate the script.

For what is worth, trying the exact code in Chrome on desktop leads to this error :

```
light-promise.js?ab:18
Uncaught ReferenceError: process is not defined
```

Adding the keyword var for process fix this :

``` js
var process;
var $nextTick = process ? process.nextTick : setTimeout;
```

**Comment by astik** [comment]
After changing all of previous things, a new problem appears :

``` js
root.Promise != null && (root.Promise = factory(root));
```

Android wechat is saying that root.Promise is undefined, so that "root.Promise != null" is false and light-promise does not get initialized.
This is not a problem on browser that support native Promise (they don't need light-promise anyway) but for older browser, the polyfill is not initializing.

Changing the JS code to this code fix the issue :

``` js
!root.Promise && (root.Promise = factory(root));
```

As I'm not a ES6 programmer (i know, i should =P) i can't help (yet) on the source code.

**Comment by dracupid** [comment]
Thank you for your patience. It's my fault that I didn't test this project in browser environment. I will fix it soon.

**Comment by dracupid** [comment]
well, undefined == null should be true, actually.

**Comment by dracupid** [comment]
As for `root.Promise != null`, I feel so sorry. I reversed the equation for debugging, but forgot to change it back.

**Comment by dracupid** [comment]
Sorry for wasting your time on these stupid bugs and provided such detailed description. I have published v0.1.1 to fix these.

**Comment by astik** [comment]
v0.1.1 is working perfectly, thanks !
